Index: WALinuxAgent/azurelinuxagent/agent.py
===================================================================
--- WALinuxAgent.orig/azurelinuxagent/agent.py
+++ WALinuxAgent/azurelinuxagent/agent.py
@@ -254,6 +254,8 @@ def parse_args(sys_args): # pylint: disa
             cmd = "deprovision+user"
         elif re.match("^([-/]*)deprovision", arg):
             cmd = "deprovision"
+        elif re.match("^([-/]*)provision", arg):
+            cmd = "provision"
         elif re.match("^([-/]*)daemon", arg):
             cmd = "daemon"
         elif re.match("^([-/]*)start", arg):
@@ -305,7 +307,7 @@ def usage():
     s  = "\n" # pylint: disable=C0103
     s += ("usage: {0} [-verbose] [-force] [-help] " # pylint: disable=C0103
            "-configuration-path:<path to configuration file>" 
-           "-deprovision[+user]|-register-service|-version|-daemon|-start|"
+           "-provision|-deprovision[+user]|-register-service|-version|-daemon|-start|"
            "-run-exthandlers|-show-configuration|-collect-logs [-full]"
            "").format(sys.argv[0])
     s += "\n" # pylint: disable=C0103
