Index: WALinuxAgent/azurelinuxagent/agent.py
===================================================================
--- WALinuxAgent.orig/azurelinuxagent/agent.py
+++ WALinuxAgent/azurelinuxagent/agent.py
@@ -169,6 +169,11 @@ class Agent(object):
         update_handler = get_update_handler()
         update_handler.run(debug)
 
+    def resourcedisk(self):
+        from azurelinuxagent.daemon.resourcedisk import get_resourcedisk_handler
+        resourcedisk_handler = get_resourcedisk_handler()
+        resourcedisk_handler.run()
+
     def show_configuration(self):
         configuration = conf.get_configuration()
         for k in sorted(configuration.keys()):
@@ -220,6 +225,8 @@ def main(args=[]): # pylint: disable=R09
                 agent.daemon()
             elif command == "run-exthandlers":
                 agent.run_exthandlers(debug)
+            elif command == "resourcedisk":
+                agent.resourcedisk()
             elif command == "show-configuration":
                 agent.show_configuration()
             elif command == "collect-logs":
@@ -264,6 +271,8 @@ def parse_args(sys_args): # pylint: disa
             cmd = "register-service"
         elif re.match("^([-/]*)run-exthandlers", arg):
             cmd = "run-exthandlers"
+        elif re.match("^([-/]*)resourcedisk", arg):
+            cmd = "resourcedisk"
         elif re.match("^([-/]*)version", arg):
             cmd = "version"
         elif re.match("^([-/]*)verbose", arg):
@@ -308,7 +317,7 @@ def usage():
     s += ("usage: {0} [-verbose] [-force] [-help] " # pylint: disable=C0103
            "-configuration-path:<path to configuration file>" 
            "-provision|-deprovision[+user]|-register-service|-version|-daemon|-start|"
-           "-run-exthandlers|-show-configuration|-collect-logs [-full]"
+           "-run-exthandlers|-resourcedisk|-show-configuration|-collect-logs [-full]"
            "").format(sys.argv[0])
     s += "\n" # pylint: disable=C0103
     return s
