From: Bastian Blank <bastian.blank@credativ.de>
Date: Wed, 4 Jan 2017 18:05:59 +0100
Subject: Add provision command to agent

---
 azurelinuxagent/agent.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/azurelinuxagent/agent.py b/azurelinuxagent/agent.py
index 6e5c92d..d660b4d 100644
--- a/azurelinuxagent/agent.py
+++ b/azurelinuxagent/agent.py
@@ -200,6 +200,8 @@ def parse_args(sys_args):
             cmd = "deprovision+user"
         elif re.match("^([-/]*)deprovision", a):
             cmd = "deprovision"
+        elif re.match("^([-/]*)provision", a):
+            cmd = "provision"
         elif re.match("^([-/]*)daemon", a):
             cmd = "daemon"
         elif re.match("^([-/]*)start", a):
@@ -246,7 +248,7 @@ def usage():
     s  = "\n"
     s += ("usage: {0} [-verbose] [-force] [-help] "
            "-configuration-path:<path to configuration file>"
-           "-deprovision[+user]|-register-service|-version|-daemon|-start|"
+           "-provision|-deprovision[+user]|-register-service|-version|-daemon|-start|"
            "-run-exthandlers|-show-configuration]"
            "").format(sys.argv[0])
     s += "\n"
