Index: WALinuxAgent/azurelinuxagent/common/osutil/default.py
===================================================================
--- WALinuxAgent.orig/azurelinuxagent/common/osutil/default.py
+++ WALinuxAgent/azurelinuxagent/common/osutil/default.py
@@ -388,9 +388,9 @@ class DefaultOSUtil(object): # pylint: d
             return
 
         if expiration is not None:
-            cmd = ["useradd", "-m", username, "-e", expiration]
+            cmd = ["useradd", "-m", username, "-s", "/bin/bash", "-e", expiration]
         else:
-            cmd = ["useradd", "-m", username]
+            cmd = ["useradd", "-m", username, "-s", "/bin/bash"]
         
         if comment is not None:
             cmd.extend(["-c", comment])
